api/net/ip4/icmp4_common.hpp

```diff
@@ -0,0 +1,183 @@
+// This file is a part of the IncludeOS unikernel - www.includeos.org
+//
+// Copyright 2017 Oslo and Akershus University College of Applied Sciences
+// and Alfred Bratterud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#ifndef NET_IP4_ICMP4_COMMON_HPP
+#define NET_IP4_ICMP4_COMMON_HPP
+
+namespace net {
+
+  /** ICMP4 Codes so that UDP and IP4 can use these */
+  namespace icmp4 {
+
+  // ICMP types
+  enum class Type : uint8_t {
+    ECHO_REPLY,
+    DEST_UNREACHABLE    = 3,
+    REDIRECT            = 5,
+    ECHO                = 8,
+    TIME_EXCEEDED       = 11,
+    PARAMETER_PROBLEM   = 12,
+    TIMESTAMP           = 13,
+    TIMESTAMP_REPLY     = 14,
+    NO_REPLY            = 100   // Custom: Type in ICMP_packet if no ping reply received
+  };
+
+  namespace code {
+
+    enum class Dest_unreachable : uint8_t {
+      NET,
+      HOST,
+      PROTOCOL,
+      PORT,
+      FRAGMENTATION,
+      SRC_ROUTE,
+      NET_UNKNOWN,          // RFC 1122
+      HOST_UNKNOWN,
+      SRC_HOST_ISOLATED,
+      NET_PROHIBITED,
+      HOST_PROHIBITED,
+      NET_FOR_TOS,
+      HOST_FOR_TOS
+    };
+
+    enum class Redirect : uint8_t {
+      NET,
+      HOST,
+      TOS_NET,
+      TOS_HOST
+    };
+
+    enum class Time_exceeded : uint8_t {
+      TTL,
+      FRAGMENT_REASSEMBLY
+    };
+
+    enum class Parameter_problem : uint8_t {
+      POINTER_INDICATES_ERROR,
+      REQUIRED_OPT_MISSING      // RFC 1122
+    };
+
+  } // < namespace code
+
+  static std::string __attribute__((unused)) get_type_string(Type type) {
+    return [&type] () {
+      switch (type) {
+        case Type::ECHO:
+          return "ECHO (8)";
+        case Type::ECHO_REPLY:
+          return "ECHO REPLY (0)";
+        case Type::DEST_UNREACHABLE:
+          return "DESTINATION UNREACHABLE (3)";
+        case Type::REDIRECT:
+          return "REDIRECT (5)";
+        case Type::TIME_EXCEEDED:
+          return "TIME EXCEEDED (11)";
+        case Type::PARAMETER_PROBLEM:
+          return "PARAMETER PROBLEM (12)";
+        case Type::TIMESTAMP:
+          return "TIMESTAMP (13)";
+        case Type::TIMESTAMP_REPLY:
+          return "TIMESTAMP REPLY (14)";
+        case Type::NO_REPLY:
+          return "NO REPLY";
+      };
+    }();
+  }
+
+  static std::string __attribute__((unused)) get_code_string(Type type, uint8_t code) {
+    return [&type, &code] () {
```

Comment:
Same as above. 
Maybe have helper functions for the nested switch cases? It also could make sense to move this to a cpp file.